flytekit/common/tasks/presto_task.py

```diff
@@ -0,0 +1,141 @@
+        :param Text schema: The schema to set for the given Presto query
+        :param dict[Text,flytekit.common.types.base_sdk_types.FlyteSdkType] task_inputs: Optional inputs to the Presto task
+        :param bool discoverable:
+        :param Text discovery_version: String describing the version for task discovery purposes
+        :param int retries: Number of retries to attempt
+        :param datetime.timedelta timeout:
+        """
+
+        # Set as class fields which are used down below to configure implicit
+        # parameters
+        self._routing_group = routing_group or ""
+        self._catalog = catalog or ""
+        self._schema = schema or ""
+
+        metadata = _task_model.TaskMetadata(
+            discoverable,
+            # This needs to have the proper version reflected in it
+            _task_model.RuntimeMetadata(
+                _task_model.RuntimeMetadata.RuntimeType.FLYTE_SDK, "1.0.0",
+                "python"),
+            timeout or _datetime.timedelta(seconds=0),
+            _literals.RetryStrategy(retries),
+            discovery_version,
+            "This is deprecated!"
+        )
+
+        presto_query = PrestoQuery(
+            routing_group=routing_group or "",
+            catalog=catalog or "",
+            schema=schema or "",
+            statement=query
+        )
+
+        # Here we set the routing_group, catalog, and schema as implicit
+        # parameters for caching purposes
+        i = interface.TypedInterface(
+            {
+                "implicit_routing_group": _interface_model.Variable(
+                    type=_types.LiteralType(simple=_types.SimpleType.STRING),
+                    description="The routing group set as an implicit input"
+                ),
+                "implicit_catalog": _interface_model.Variable(
+                    type=_types.LiteralType(simple=_types.SimpleType.STRING),
+                    description="The catalog set as an implicit input"
+                ),
+                "implicit_schema": _interface_model.Variable(
+                    type=_types.LiteralType(simple=_types.SimpleType.STRING),
+                    description="The schema set as an implicit input"
+                )
+            },
+            {
+                # Set the schema for the Presto query as an output
+                "results": _interface_model.Variable(
+                    type=_types.LiteralType(schema=output_schema),
+                    description="The schema for the Presto query"
+                )
+            })
+
+        super(SdkPrestoTask, self).__init__(
+            _constants.SdkTaskType.PRESTO_TASK,
+            metadata,
+            i,
+            _MessageToDict(presto_query.to_flyte_idl()),
+        )
+
+        # Set user provided inputs
+        task_inputs(self)
+
+    # Override method in order to set the implicit inputs
+    def __call__(self, *args, **kwargs):
```

Comment:
if this task can't accept additional inputs when creating a node, we should just make the method `def __call__(self)`